model/src/main/proto/platform_parameter.proto

```diff
@@ -0,0 +1,37 @@
+syntax = "proto3";
+
+package model;
+
+option java_package = "org.oppia.android.app.model";
+option java_multiple_files = true;
+
+// It will correspond to a single Platform Parameter Value.
+message PlatformParameter {
+    // Value of a parameter can be of any one of the following type.
+    oneof value {
+        // Boolean type parameter value.
+        bool boolean = 2;
+        // Integer type parameter value.
+        int32 integer = 3;
+        // String type parameter value.
+        string string = 4;
+    }
+}
+
+// Storage format of platform parameters while the app is running.
+message PlatformParameterMap{
+    map<string, PlatformParameter> parameter = 1;
+}
+
+// Pair of Platform Parameter Value and Name.
```

Comment:
Why not always pair a name with platform parameters? They are expected to have one, and for compile-time only parameters we can have a special case for when the name is undefined (empty), e.g.:

```proto
// The name of the platform parameter, or empty if the parameter is only defined locally.
string name = 1;
...
```

Reply:
We didn't included the name of the Parameter in this proto class becuase when it was being used in the ParameterMap so we had a repetition of names, once as a key and second inside the object being stored as value. /cc @vinitamurthi

Reply:
Ah I see. How important is the map? An alternate approach might be just to store a repeated list of the parameters and build the map at runtime.

Reply:
Actually the map is formed only at the runtime. It has a keyset that corresponds to all the platform names. The value set of this map contains PlatformParameter objects which currently wrap only the value of a particular parameter. But if we would have also stored the names inside PlatformParameter objects then there would have been unnecessary repititon of the names in the key-value pairs of map. We avoided this problem by using a complete new proto (PlatformParameterWithName) class which contains both the name and value of Platform Parameters.

Reply:
Hmm, I think I might be missing something. Is this the compile-time name of the parameter, or the server name? If the former, it seems wrong not to include it in the PlatformParameter since that is, conceptually, a parameter. I see the point that it would be duplicated, but I don't see that it's an actual issue here. Having access to the parameter name in PlatformParameter seems like a good design, and to some extent expected. PlatformParameter providing a value without the context of the name is a bit confusing.

Or am I missing some key design aspect of this?

@vinitamurthi in case you have thoughts.

Reply:
These are simply Parameter names as we are not using different Proto structure based on whether it is a compile time or runtime parameter.

Reply:
I think its okay if we add the name in the internal message. We would be duplicating the name but I think that's not a big issue